brilirs/src/check.rs

```diff
@@ -400,7 +497,11 @@ fn type_check_instruction<'a>(
         Some(t) => {
           check_num_args(1, args)?;
           let ty0 = get_type(env, 0, args)?;
-          check_asmt_type(t, ty0)
+          if let Some(promise_t) = promise {
```

Comment:
It's unclear to me why the `promise` variable needs to be threaded through as an argument to `type_check_instruction`. Isn't this information available inside of `t` here? If so, that would be simpler